Expensify/App
# [IOU] Create IOUModal with steps

CC @iwiznia -- no backend logic to review here, but FYI

### Details
Implement the base IOUModal as a right docked modal with controlled step components, navigation, and Onyx driven-loading state. The individual steps will be built out by external contributors based on this groundwork.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/154607

### Tests

First, apply the following diff, or temporarily swap the CreateMenu.js routes from `NEW_CHAT > IOU_REQUEST_MONEY` & `NEW_GROUP > IOU_BILL_SPLIT`
``` 
diff --git a/src/components/CreateMenu.js b/src/components/CreateMenu.js
index de365afe..69e67a1a 100644
--- a/src/components/CreateMenu.js
+++ b/src/components/CreateMenu.js
@@ -55,13 +55,13 @@ class CreateMenu extends PureComponent {
         const menuItemData = [
             {
                 icon: ChatBubble,
-                text: 'New Chat',
-                onPress: () => this.setOnModalHide(() => redirect(ROUTES.NEW_CHAT)),
+                text: 'Request Money',
+                onPress: () => this.setOnModalHide(() => redirect(ROUTES.IOU_REQUEST_MONEY)),
             },
             {
+                text: 'Split Bill',
                 icon: Users,
-                text: 'New Group',
-                onPress: () => this.setOnModalHide(() => redirect(ROUTES.NEW_GROUP)),
+                onPress: () => this.setOnModalHide(() => redirect(ROUTES.IOU_BILL_SPLIT)),
             },
         ].map(item => ({
             ...item,

```
_*Make sure to include the last newline in the diff!_
_______

**Test Modal navigation**
1) Open the global create menu and select either option
2) The Modal should display as a right docked Modal on Web
3) The cross button should close the Modal
4) You should be able to navigate forwards through the steps with the buttons
5) You should be able to navigate backwards through the steps with the arrow
6) The arrow button should not display on the first step
7) Each step should display a title: 'Amount', 'Participants', and 'Confirm'

**Test that we distinguish between Request Money/Split Bill flows** 
1) Launch the Modal via 'Request Money'
2) Navigate to the Participants step
3) Confirm that `// single` is displayed above the loading spinner
4) Launch the Modal via 'Split Bill'
5) Navigate to the Participants step
6) Confirm that `// group` is displayed above the loading spinner

**Test that Modal is opened as a new instance**
1) Navigate to step 2 or 3 of the Modal
2) Close the Modal
3) Reopen the Modal
4) Modal state should be reset, you should start on step 1 again

**Test that Modal steps are NOT opened as a new instance**
1) Notice the loading spinner on the first step
2) Wait 1.6 seconds for it to disappear ([due to this hardcoded delay](https://github.com/Expensify/Expensify.cash/blob/f09b4da3a3a9c7232a0a74d75f2b1a42488fb7e8/src/libs/actions/IOU.js#L11-L13))
3) Navigate away from step 1 with the 'Next' button
4) Navigate back to step 1 with the back arrow
5) The loading spinner **should NOT** reappear, proving that the step Component state is persisted

**Test that refreshing the page doesn't cause an error**
1) On web, open the modal
2) Quickly refresh the page before the loading spinner disappears (1.6 seconds)
3) You should not get stuck in a permanent loading state
4) Close the Modal
5) Open a new Modal on web
6) Navigate to step 2 or 3
7) Refresh the page
8) You should be returned to the 1st page

### Tested On


### Screenshots

#### Web
![IOUWeb1](https://user-images.githubusercontent.com/10736861/109992247-fe9c6100-7d02-11eb-9af5-cfe320bd7d5d.png)
![IOUWeb2](https://user-images.githubusercontent.com/10736861/109992266-00febb00-7d03-11eb-9cf4-a66475fedfa7.png)
![IOUWeb3](https://user-images.githubusercontent.com/10736861/109992276-02c87e80-7d03-11eb-829c-c2a940ea9b80.png)

#### Mobile Web
![IOUmobileWeb](https://user-images.githubusercontent.com/10736861/109995914-a10a1380-7d06-11eb-8a4f-60550d5c6a66.png)

#### Desktop
<img width="972" alt="IOUDesktop" src="https://user-images.githubusercontent.com/10736861/109992457-32778680-7d03-11eb-98c8-dffebcc80759.png">

#### iOS
![IOUiOS](https://user-images.githubusercontent.com/10736861/109995529-47a1e480-7d06-11eb-929e-ff5c02baa0d5.png)

#### Android
![IOUAndroid](https://user-images.githubusercontent.com/10736861/109994166-f2b19e80-7d04-11eb-8f0c-cee2375c9ae3.png)
